### PR TITLE
fix: (CS-32183) onEntryChange calling api 2 times

### PR DESCRIPTION
**Pull request by contentstackMridul**
1. added new function onLiveEdit which will call the management API only once
2. added optional param to onEntryChange

### CODE REVIEW

File: src/live-preview.ts [hunk 1]
```diff
@@ -333,9 +333,9 @@ export default class LivePreview {
 
     // Request parent for data sync when document loads
     private requestDataSync() {
-        this.handleUserChange({
-            live_preview: "init", // this is the hash of the live previewd
-        });
+        // this.handleUserChange({
```
Comment (Deepak-Kharah) [inline src/live-preview.ts:336]: Do not keep dead code. Remove it and add a commit message describing why

File: src/contentstack-live-preview-HOC.ts [hunk 2]
```diff
@@ -55,10 +55,16 @@ export class ContentstackLivePreview {
...
-
+/**
+ * @type {function}
+ * @param onChangeCallback A function param to fetch the data from contentstack database
+ * @param config An optional object param, pass {skipInitRun:Boolean} to skip init call to onChangeCallback
+ * @returns Subscribed Callback UID 
+ */
     static onEntryChange(
-        onChangeCallback: OnEntryChangeCallback
+        onChangeCallback: OnEntryChangeCallback,config?:{skipInitRun?:boolean}
```
Comment (Deepak-Kharah) [inline src/contentstack-live-preview-HOC.ts:65]: Let's move the config type into a separate interface and use it here.

**Review by Deepak-Kharah — CHANGES_REQUESTED** [review]
I have added some comments-- let's fix them. Moreover, let's add the unit test cases for `onEntryChange()` with the new flag and `onLiveEdit()`

File: src/live-preview.ts (hunk 1 above)
Comment (contentstackMridul) [reply src/live-preview.ts]: done

File: src/contentstack-live-preview-HOC.ts (hunk 2 above)
Comment (contentstackMridul) [reply src/contentstack-live-preview-HOC.ts]: done

File: src/utils/types.ts [hunk 3]
```diff
@@ -101,4 +101,8 @@ export declare interface IDocWithScriptMessage
 
 export declare type OnEntryChangeCallback = () => void;
 
+export declare type OnEntryChangeConfig = {
+    skipInitRun?: boolean;
```
Comment (Deepak-Kharah) [inline src/utils/types.ts:105]: let's rename it to `skipInitialRender`

**Review by Deepak-Kharah — APPROVED** [review]
The changes look good. The test should be enough. Let's cover E2E as well

**Pull request merged**